pkg/auth/providers/keycloakoidc/keycloak_client.go

```
@@ -0,0 +1,179 @@
+	search := URLEncoded(searchURL)
+
+	b, err := k.getFromKeyCloak(accessToken, search)
+	if err != nil {
+		// no match on search returns an error. do not log
+		return nil, nil
+	}
+	var searchResults []Group
+	if err := json.Unmarshal(b, &searchResults); err != nil {
+		logrus.Errorf("[keycloak oidc]: received error unmarshalling search results, err: %v", err)
+		return searchResults, err
+	}
+	return searchResults, nil
+}
+
+func (k *KClient) getFromKeyCloakByID(principalID, accessToken, searchType string, config *v32.OIDCConfig) (Account, error) {
+	searchURL := fmt.Sprintf("%s/%s/%s", k.getSearchURL(config.Issuer), searchType, principalID)
+	search := URLEncoded(searchURL)
+	b, err := k.getFromKeyCloak(accessToken, search)
+	if err != nil {
+		return Account{}, nil
+	}
+	var searchResult Account
+	if err := json.Unmarshal(b, &searchResult); err != nil {
+		logrus.Errorf("[keycloak oidc]: received error unmarshalling search results, err: %v", err)
+		return searchResult, err
+	}
+	return searchResult, nil
+
+}
+func (k *KClient) getSearchURL(issuer string) string {
+	splitIssuer := strings.SplitAfter(issuer, "/auth/")
+	return fmt.Sprintf(
+		"%sadmin/%s",
+		splitIssuer[0],
+		splitIssuer[1],
+	)
+}
+
+//URLEncoded encodes the string
+func URLEncoded(str string) string {
+	u, err := url.Parse(str)
+	if err != nil {
+		logrus.Errorf("Error encoding the url: %s, error: %v", str, err)
+		return str
+	}
+	return u.String()
+}
+
+func (k *KClient) getFromKeyCloak(accessToken string, url string) ([]byte, error) {
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Add("Authorization", "token "+accessToken)
+	req.Header.Add("Accept", "application/json")
+	resp, err := k.httpClient.Do(req)
+	if err != nil {
+		logrus.Errorf("[keycloak oidc] received error from keycloak: %v", err)
+		return nil, err
+	}
+	defer resp.Body.Close()
+	// Check the status code
+	switch resp.StatusCode {
+	case 200:
+	case 201:
```

So we only accept 200/201 and otherwise fail ? Thats somewhat strict but it might work. Does keycloak specifically talk about those 2 codes ? I'm worried if you get a 301 in the middle you'll end up in the error section, but perhaps that case would be incorrectly configured.

---

From what I have seen, when a search returns nothing, it is a 401 not found. Either way I am ignoring the error messages later so if this seems too strict I can remove the case
```		
b, err := k.getFromKeyCloak(accessToken, search)
if err != nil {
  // no match on search returns an error. do not log
  return nil, nil
}
```

---

We need to handler errors properly

---

errors are now being handled where this func is called